dataset_builders/pie/comagc/comagc.py

```diff
@@ -0,0 +1,198 @@
+                "relations": "binary_relations",
+            }
+        }
+
+    def _generate_document(self, example, **kwargs):
+        return example_to_document(example)
+
+    def _generate_example(self, document: Document, **kwargs) -> Dict[str, Any]:
+        return document_to_example(document)
+
+
+def get_relation(example):
+    """Simple rule-based function to determine the relation between the gene and the cancer."""
+    # https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-323/tables/3
+
+    rules = [
+        {
+            "CGE": "increased",
+            "CCS": "normalTOcancer",
+            "IGE": "*",
+            "PT": "causality",
+            "Gene class": "oncogene",
+        },
+        {
+            "CGE": "decreased",
+            "CCS": "cancerTOnormal",
+            "IGE": "unidentifiable",
+            "PT": "causality",
+            "Gene class": "oncogene",
+        },
+        {
+            "CGE": "decreased",
+            "CCS": "cancerTOnormal",
+            "IGE": "up-regulated",
+            "PT": "*",
+            "Gene class": "oncogene",
+        },
+        {
+            "CGE": "decreased",
+            "CCS": "normalTOcancer",
+            "IGE": "*",
+            "PT": "causality",
+            "Gene class": "tumor suppressor gene",
+        },
+        {
+            "CGE": "increased",
+            "CCS": "cancerTOnormal",
+            "IGE": "unidentifiable",
+            "PT": "causality",
+            "Gene class": "tumor suppressor gene",
+        },
+        {
+            "CGE": "increased",
+            "CCS": "cancerTOnormal",
+            "IGE": "down-regulated",
+            "PT": "*",
+            "Gene class": "tumor suppressor gene",
+        },
+        {
+            "CGE": "*",
+            "CCS": "normalTOcancer",
+            "IGE": "*",
+            "PT": "observation",
+            "Gene class": "biomarker",
+        },
+        {
+            "CGE": "*",
+            "CCS": "cancerTOnormal",
+            "IGE": "unidentifiable",
+            "PT": "observation",
+            "Gene class": "biomarker",
+        },
+        {
+            "CGE": "decreased",
+            "CCS": "cancer->cancer",
+            "IGE": "up-regulated",
+            "PT": "observation",
+            "Gene class": "biomarker",
+        },
+        {
+            "CGE": "increased",
+            "CCS": "cancer->cancer",
+            "IGE": "down-regulated",
+            "PT": "observation",
+            "Gene class": "biomarker",
+        },
+    ]
+
+    for rule in rules:
+        if (
+            (rule["CGE"] == "*" or example["CGE"] == rule["CGE"])
+            and (rule["CCS"] == "*" or example["CCS"] == rule["CCS"])
+            and (rule["IGE"] == "*" or example["IGE"] == rule["IGE"])
+            and (rule["PT"] == "*" or example["PT"] == rule["PT"])
+        ):
+            return rule["Gene class"]
+
+    return "unidentified"
```

Comment:
Did you make this up or is this an official label? Anything that is not official (in the original data), should be avoided. If this is not possible, it should be at least marked as such and explained why it is necessary.

Reply:
I invented it, as there are entries, where no rule does apply and therefore no _gene-class_ relation can be assigned. (At least 100 for the entire dataset of 821 examples)

Reply:
hmm ok, still not 100% clear. But maybe I need to have the full explanation how `get_relation_label` works before asking further questions.

Reply:
Added it to the README.md as well